core/api/serializers/project.py

```diff
@@ -94,3 +178,75 @@ def get_submission(self, obj):
 
     def get_sector(self, obj):
         return obj.subsector.sector.name
+
+
+class ProjectDetailsSerializer(ProjectListSerializer):
+    """
+    ProjectSerializer class
+    """
+
+    submission = ProjectSubmissionSerializer()
+    ods_odp = ProjectOdsOdpSerializer(many=True)
+    agency_id = serializers.IntegerField(required=True)
+    country_id = serializers.IntegerField(required=True)
+    subsector_id = serializers.IntegerField(required=True)
+    project_type_id = serializers.IntegerField(required=True)
+
+    class Meta:
+        model = Project
+        fields = ProjectListSerializer.Meta.fields + [
+            "description",
+            "country_id",
+            "agency_id",
+            "national_agency",
+            "subsector_id",
+            "project_type_id",
+            "approval_meeting_no",
+            "submission",
+            "ods_odp",
+        ]
+
+    def validate_country_id(self, value):
+        country = Country.objects.filter(id=value).first()
+        if not country:
+            raise serializers.ValidationError(f"Invalid country id {value}")
+        return value
+
+    def validate_agency_id(self, value):
+        agency = Agency.objects.filter(id=value).first()
+        if not agency:
+            raise serializers.ValidationError(f"Invalid agency id {value}")
+        return value
+
+    def validate_subsector_id(self, value):
+        subsector = ProjectSubSector.objects.filter(id=value).first()
+        if not subsector:
+            raise serializers.ValidationError(f"Invalid subsector id {value}")
+        return value
+
+    def validate_project_type_id(self, value):
+        project_type = ProjectType.objects.filter(id=value).first()
+        if not project_type:
+            raise serializers.ValidationError(f"Invalid project type id {value}")
+        return value
+
+    def create(self, validated_data):
+        submission_data = validated_data.pop("submission")
+        ods_odp_data = validated_data.pop("ods_odp")
+
+        if submission_data:
+            status = ProjectStatus.objects.get(code="NEWSUB")
+            validated_data["status_id"] = status.id
+        else:
+            status = ProjectStatus.objects.get(code="NEW")
+            validated_data["status_id"] = status.id
+
+        # create project
+        project = Project.objects.create(**validated_data)
+        # create submission
+        if submission_data:
+            ProjectSubmission.objects.create(project=project, **submission_data)
+        # create ods_odp
+        for ods_odp in ods_odp_data:
+            ProjectOdsOdp.objects.create(project=project, **ods_odp)
+        return project
```

Comment:
By default in Django requests are not atomic. But can be enabled with the [ATOMIC_REQUESTS](https://docs.djangoproject.com/en/4.2/ref/settings/#atomic-requests) setting. 

In general it's better to not enable this as most requests do not need to be atomic, so there is a speed penalty for processing the superfluous transactions. 

---

In this case we want to wrap this create with a transaction, because if any of these insert operations fail we will end up with an incomplete state in the DB. It will be preferable to simply fail the the request (hence rolling back the transaction) and have the user try again.